### PR TITLE
Minor fixes to fs/file.cpp

### PR DESCRIPTION
**Pull request by PabloMK7**
Changes `std::make_unique` to `std::make_unique_for_overwrite` which makes the code cleaner while keeping the data buffer from being initialized to 0.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7262)

### CODE REVIEW

File: src/core/hle/service/fs/file.cpp [hunk 1]
```diff
@@ -79,13 +79,13 @@ void File::Read(Kernel::HLERequestContext& ctx) {
     if (!backend->AllowsCachedReads()) {
         auto& buffer = rp.PopMappedBuffer();
         IPC::RequestBuilder rb = rp.MakeBuilder(2, 2);
-        std::unique_ptr<u8*> data = std::make_unique<u8*>(static_cast<u8*>(operator new(length)));
-        const auto read = backend->Read(offset, length, *data);
+        std::unique_ptr<u8[]> data = std::make_unique_for_overwrite<u8[]>(length);
```
Comment (SachinVin) [inline src/core/hle/service/fs/file.cpp:82]: Why couldn't this be a vector?

Comment (PabloMK7) [reply src/core/hle/service/fs/file.cpp:82]: Vector initializes all its elements to 0 by default when constructed, which is unneeded as the data will be replaced afterwards.

Comment (SachinVin) [reply src/core/hle/service/fs/file.cpp:82]: Hmm, how much of an overhead is it really?

**Comment by PabloMK7** [comment]
Dropping the idea as the needed function is not available in all environments.

**Pull request closed without merging**